esp/esp/program/modules/handlers/studentregtwophase.py

```diff
@@ -148,14 +148,12 @@ def studentreg2phase(self, request, tl, one, two, module, extra, prog):
 
             if timeslot.id in timeslot_dict:
                 priority_dict = timeslot_dict[timeslot.id]
-                # (relationship, class_title) -> relationship.name
-                priority_list = sorted(priority_dict.items(), key=lambda item: item[0].name)
             else:
-                priority_list = []
+                priority_dict = {}
             temp_list = []
             for i in range(0, context['num_priority']):
```

Comment:
What's probably even better, here, is to use `relationship.displayName` as the description.  I think doing your own custom stringification is really not ideal here; much better to just sort by the `.name` but use the `.displayName`.  You might need to refactor how this works a little, but I think it will make things a lot clearer.

Reply:
Oh, I wasn't even aware of `.displayName`. In fact, it's fairly odd that that wasn't used from the get-go, especially for the user interface (the /rank_classes still uses the Priority/X format, and the main page used that format until I changed it with #2425).

I'm not totally sure I understand what you're looking for though, since it sounds like you want to go back to the sorting method that seemed bugged? Also, I'm not sure how we'd be able to accommodate empty/absent priorities without looping through the priorities that _should_ exist (via stringification) rather than just the priorities that _do_ exist.

Reply:
bump @benjaminjkraft

Reply:
Yeah, I guess it's ok to specifically look for `Priority/1` and so on to generate the priorities.  But instead of manually stringifying `Priority 1` for the display name, just use `relationship.displayName`, which on some sites is set to something even nicer like `First Choice`.

Reply:
Oh, okay. And if they are missing a priority, just use the "Priority X" text? Even if it won't match the other `displayName`s?

Reply:
Oh, I see the issue.  I think what you can do is look up the `RegistrationType` with that name, even if it's not attached to any `StudentRegistration`s for this student.  It should be guaranteed to exist.